### PR TITLE
Tests for incremental small strain and finite strain formulations for gps model

### LINKED ISSUE #7835: Tests for generalized plane strain model for incremental small strain and finite strain formulations
### Description of the enhancement or error report
Regression tests are added for incremental small and finite strain formulations of generalized plane strain model

### Identified impact
New added capabilities and tests

### PR DESCRIPTION
**Pull request by hchen139**
closes #7835

### CODE REVIEW

**Comment by hchen139** [comment]
@bwspenc @jiangwen84 @sapitts this PR is for the regression tests of generalized plane strain model using incremental small strain and finite strain formulations.

File: modules/tensor_mechanics/src/materials/ComputeAxisymmetricRZFiniteStrain.C [hunk 1]
```diff
@@ -34,7 +34,7 @@ ComputeAxisymmetricRZFiniteStrain::initialSetup()
 }
 
 Real
-ComputeAxisymmetricRZFiniteStrain::computeDeformGradZZ()
```
Comment (bwspenc) [inline modules/tensor_mechanics/src/materials/ComputeAxisymmetricRZFiniteStrain.C]: I don't think these functions make any sense for axisymmetric cases. The out of plane strain (theta theta) is computed as a function of the radial displacement.

File: modules/tensor_mechanics/src/materials/ComputeAxisymmetricRZIncrementalStrain.C [hunk 2]
```diff
@@ -33,7 +33,7 @@ ComputeAxisymmetricRZIncrementalStrain::initialSetup()
 }
 
 Real
-ComputeAxisymmetricRZIncrementalStrain::computeDeformGradZZ()
```
Comment (bwspenc) [inline modules/tensor_mechanics/src/materials/ComputeAxisymmetricRZIncrementalStrain.C]: Same as the comment about the other axisymmetric case.

File: modules/tensor_mechanics/src/materials/Compute2DFiniteStrain.C [hunk 3]
```diff
@@ -37,9 +37,9 @@ Compute2DFiniteStrain::computeProperties()
     RankTwoTensor A((*_grad_disp[0])[_qp], (*_grad_disp[1])[_qp], (*_grad_disp[2])[_qp]); //Deformation gradient
     RankTwoTensor Fbar((*_grad_disp_old[0])[_qp], (*_grad_disp_old[1])[_qp], (*_grad_disp_old[2])[_qp]); //Old Deformation gradient
 
-    // Compute the deformation gradient (2,*) value for plane strain, generalized plane strain, or axisymmetric problems
-    A(2,2) = computeDeformGradZZ();
-    Fbar(2,2) = computeDeformGradZZold();
+    // Compute the displacement gradient (2,2) value for plane strain, generalized plane strain, or axisymmetric problems
```
Comment (bwspenc) [inline modules/tensor_mechanics/src/materials/Compute2DFiniteStrain.C]: We should discuss the name of this function.  It seems to me like the old name was fine, but you must have had a reason for changing it, so I'd like to hear your thoughts.

**Review by bwspenc — CHANGES_REQUESTED** [review]
Overall, looks good.  Just a couple of minor issues to work out as noted and then I think it's ready to go.

File: modules/tensor_mechanics/src/materials/Compute2DFiniteStrain.C (hunk 3 above)
Comment (bwspenc) [inline modules/tensor_mechanics/src/materials/Compute2DFiniteStrain.C]: OK, we discussed this and I agree.

File: modules/tensor_mechanics/src/materials/ComputeAxisymmetricRZFiniteStrain.C (hunk 1 above)
Comment (bwspenc) [inline modules/tensor_mechanics/src/materials/ComputeAxisymmetricRZFiniteStrain.C]: Never mind about this as well.  After looking at more context, I see what's going on here, and this is fine.

File: modules/tensor_mechanics/src/materials/ComputeAxisymmetricRZIncrementalStrain.C (hunk 2 above)
Comment (bwspenc) [inline modules/tensor_mechanics/src/materials/ComputeAxisymmetricRZIncrementalStrain.C]: Never mind here as well.

**Review by bwspenc — APPROVED** [review]

**Comment by bwspenc** [comment]
@aeslaughter I'm good with these changes and it's ready to go as far as I'm concerned.

**Review by aeslaughter — APPROVED** [review]
@bwspenc Approves these changes.

**Pull request merged**